databricks_cli/workspace/types.py

```diff
@@ -30,23 +30,21 @@ class WorkspaceLanguage(object):
     SQL = 'SQL'
     R = 'R'
     ALL = [SCALA, PYTHON, SQL, R]
-    EXTENSIONS = ['.scala', '.py', '.sql', '.SQL', '.r', '.R']
+    EXTENSIONS = ['.scala', '.py', '.sql', '.SQL', '.r', '.R', '.ipynb']
 
     @classmethod
-    def to_language(cls, path):
-        ext = cls.get_extension(path)
+    def to_language_and_format(cls, path):
+        ext = cls.get_extension(path).lower()
         if ext == '.scala':
-            return cls.SCALA
+            return (cls.SCALA, WorkspaceFormat.SOURCE)
         elif ext == '.py':
-            return cls.PYTHON
+            return (cls.PYTHON, WorkspaceFormat.SOURCE)
         elif ext == '.sql':
-            return cls.SQL
-        elif ext == '.SQL':
-            return cls.SQL
+            return (cls.SQL, WorkspaceFormat.SOURCE)
         elif ext == '.r':
-            return cls.R
-        elif ext == '.R':
```

Comment:
Let's not remove this extension.

Reply:
I just added a `.lower()` to the extension to reduce the condition checks

Reply:
Gotcha.

Reply:
pylint was complaining about too many returns in that method after I added the jupyter one. However I'm getting 0 messages from pylint now but the Travis build still fails?

Reply:
I saw the same behavior. It's because Travis is using a newer verison of pylint. Locally I have pylint (1.7.2) but on Travis it's pylint (1.8.1).